geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -773,14 +773,14 @@ void doNormalMessage() {
     }
     Message message;
     message = BaseCommand.readRequest(this);
-    synchronized (serverConnectionCollection) {
-      if (serverConnectionCollection.isTerminating) {
-        // Client is being disconnected, don't try to process message.
-        this.processMessages = false;
-        return;
-      }
-      serverConnectionCollection.connectionsProcessing.incrementAndGet();
+    if (serverConnectionCollection.isTerminating) {
```

Comment:
It seems like this should be one method on ServerConnectionCollection that can both test isTerminating and increment the processing count. You could call it incrementProcessing() and if it returns false set this.processMessages to false and return. I'd like to see the ServerConnectionCollection instance variable made private instead of other classes (like this one) access them.